modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/internal/HttpHeadersAuthenticatorUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http.netty4.internal;
+
+import io.netty.handler.codec.http.DefaultHttpRequest;
+import io.netty.handler.codec.http.HttpMessage;
+import io.netty.handler.codec.http.HttpRequest;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.http.HttpHeadersValidationException;
+import org.elasticsearch.http.HttpPreRequest;
+import org.elasticsearch.http.netty4.Netty4HttpHeaderValidator;
+import org.elasticsearch.http.netty4.Netty4HttpRequest;
+import org.elasticsearch.rest.RestRequest;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.http.netty4.Netty4HttpRequest.getHttpHeadersAsMap;
+import static org.elasticsearch.http.netty4.Netty4HttpRequest.translateRequestMethod;
+
+/**
+ * Provides utilities for hooking into the netty pipeline and authenticate each HTTP request's headers.
+ * See also {@link Netty4HttpHeaderValidator}.
+ */
+public final class HttpHeadersAuthenticatorUtils {
+
+    // utility class
+    private HttpHeadersAuthenticatorUtils() {}
+
+    /**
+     * Supplies a netty {@code ChannelInboundHandler} that runs the provided {@param authenticator} on the HTTP request headers.
```

Comment:
nit: javadoc says authenticator but param is called validator

Reply:
Handled in [0d03f6e](https://github.com/elastic/elasticsearch/pull/95112/commits/0d03f6e324ad98f82c2a56d13320d22c2300acfe) .